nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/datasetbrowser/RemoteDatasetBrowserTabController.java

```diff
@@ -0,0 +1,227 @@
+package org.nmrfx.analyst.gui.datasetbrowser;
+
+import com.jcraft.jsch.JSchException;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import org.nmrfx.analyst.gui.AnalystApp;
+import org.nmrfx.analyst.gui.AnalystPrefs;
+import org.nmrfx.processor.datasets.vendor.NMRDataUtil;
+import org.nmrfx.processor.gui.FXMLController;
+import org.nmrfx.utilities.DatasetSummary;
+import org.nmrfx.utilities.RemoteDatasetAccess;
+import org.nmrfx.utilities.UnZipper;
+import org.nmrfx.utils.GUIUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class RemoteDatasetBrowserTabController extends DatasetBrowserTabController {
+    private static final Logger log = LoggerFactory.getLogger(RemoteDatasetBrowserTabController.class);
+    private static final String TAB_NAME = "Remote";
+    private final FileSystem fileSystem = FileSystems.getDefault();
+    private final Path pathToLocalCache = fileSystem.getPath(System.getProperty("user.home"), "NMRFx_Remote_Datasets", "data");
+    private RemoteDatasetAccess remoteDatasetAccess;
+
+    public RemoteDatasetBrowserTabController() {
+        super(TAB_NAME);
+        tableView = new DatasetBrowserTableView(true);
+        borderPane.setCenter(tableView);
+        initToolbar(true);
+        directoryTextField.setText(AnalystPrefs.getRemoteDirectory());
+    }
+
+    /**
+     * Create a RemoteDatasetAccess object and try to connect.
+     * @return True if remoteDatasetAccess is connected, otherwise false.
+     */
+    private boolean initRemoteDatasetAccess() {
+        if (remoteDatasetAccess == null) {
+            String remoteHost = AnalystPrefs.getRemoteHostName();
+            String remoteUser = AnalystPrefs.getRemoteUserName();
+
+            remoteDatasetAccess = new RemoteDatasetAccess(remoteUser, remoteHost);
+            boolean usePassword = AnalystPrefs.getUseRemotePassword();
+            if (usePassword && !remoteDatasetAccess.passwordValid()) {
+                String pw = GUIUtils.getPassword();
+                if (pw != null) {
+                    remoteDatasetAccess.setPassword(pw);
+                }
+            }
+        }
+        if (!remoteDatasetAccess.isConnected()) {
```

Comment:
Change from original behaviour, moved connection outside of object creation because if the remoteDatasetAccess failed to connect it would return false, but the next time it was called remoteDatasetAccess was no longer null so it wouldn't attempt to connect and would return true.

Reply:
Now remoteDatasetAccess is set to null if connection failed, but I still think we should check for connection separate from creation, in case the object gets disconnected somehow